mkosi/bootloader.py

```diff
@@ -585,6 +585,10 @@ def find_and_install_shim_binary(
 
                 rel = p.relative_to(context.root)
                 if (context.root / output).is_dir():
+                    # The ESP wants .efi files, not .efi.signed or .efi.signed.latest
+                    if rel.suffix and rel.suffix != ".efi":
+                        left_stem, _ = p.name.split(".", maxsplit=1)
```

Comment:
Oops, this has my typo in it from when I was testing this!
```suggestion
                        left_stem, _ = rel.name.split(".", maxsplit=1)
```